### PR TITLE
feat: add custom-tracing-subscriber example

### PR DESCRIPTION
**Pull request by orhun**

### CODE REVIEW

File: actix-web/custom-tracing-subscriber/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "custom-tracing-subscriber"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+actix-web = "4.3.1"
+shuttle-actix-web = "0.25.0"
+shuttle-runtime = { version = "0.25.0", default-features = false }
```
Comment (oddgrd) [inline actix-web/custom-tracing-subscriber/Cargo.toml:9]: Maybe a comment here too about how we disable the default features to disable the shuttle subscriber setup.

File: actix-web/custom-tracing-subscriber/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#[get("/")]
+async fn hello_world() -> &'static str {
+    let msg = "Hello World!";
+    tracing::info!("{msg}");
+    msg
+}
+
+#[shuttle_runtime::main]
+async fn actix_web() -> ShuttleActixWeb<impl FnOnce(&mut ServiceConfig) + Send + Clone + 'static> {
+    tracing_subscriber::fmt()
```
Comment (oddgrd) [inline actix-web/custom-tracing-subscriber/src/main.rs:15]: Maybe a comment about the fact that we need to write to stdout for shuttle to record our logs, and so we use the fmt subscriber which has sane defaults for applications and writes to stdout (https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/index.html). What do you think?

**Review by oddgrd — COMMENTED** [review]
Nice! I just left some notes on comments we might want to add.

File: actix-web/custom-tracing-subscriber/Cargo.toml (hunk 1 above)
Comment (orhun) [reply actix-web/custom-tracing-subscriber/Cargo.toml:10]: Done in fba420bcde99415e11dbbb03dacc78b95c72acc1

**Review by chesedo — COMMENTED** [review]
We should still update, or remove the other examples inside `/tracing`

File: tracing/custom-tracing-subscriber/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[package]
+name = "custom-tracing-subscriber"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+actix-web = "4.3.1"
+shuttle-actix-web = "0.25.0"
+# disable default features for disabling the Shuttle's default tracing subscriber
```
Comment (oddgrd) [inline tracing/custom-tracing-subscriber/Cargo.toml:9]: Maybe like this?
```suggestion
# disable default features to disable the Shuttle default tracing subscriber
```

File: tracing/custom-tracing-subscriber/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    let msg = "Hello World!";
+    tracing::info!("{msg}");
+    msg
+}
+
+#[shuttle_runtime::main]
+async fn actix_web() -> ShuttleActixWeb<impl FnOnce(&mut ServiceConfig) + Send + Clone + 'static> {
+    // Normally we write to stdout for Shuttle to record the logs (i.e. println!)
+    // But we can use a fmt subscriber instead which has sane defaults for applications and also
+    // writes to stdout.
```
Comment (oddgrd) [inline tracing/custom-tracing-subscriber/src/main.rs:17]: What do you think about this? I think we can keep it somewhat simple, and we will explain more in depth in our docs page, where we'll also link to this example.
```suggestion
    // We need to write to stdout for Shuttle to record our logs, so we use the 
    // tracing::fmt subscriber which has sane defaults for applications and also
    // writes to stdout.
```

**Review by oddgrd — COMMENTED** [review]
Nice! I have some suggestions for how we might rephrase the comments, let me know what you think.

File: tracing/custom-tracing-subscriber/src/main.rs (hunk 4 above)
Comment (orhun) [reply tracing/custom-tracing-subscriber/src/main.rs]: 6a516838dce5927185d95971109885698e0c1b26

File: tracing/custom-tracing-subscriber/Cargo.toml (hunk 3 above)
Comment (orhun) [reply tracing/custom-tracing-subscriber/Cargo.toml]: 6a516838dce5927185d95971109885698e0c1b26

**Comment by orhun** [comment]
> We should still update, or remove the other examples inside `/tracing`

Updated them in a022355599f987deb177ed27fce2407914c5b570

They both seem to need tracing initialized so I didn't disable the feature. Let me know if anything else is needed.

File: tracing/axum-logs-endpoint/src/main.rs [hunk 5]
```diff
@@ -11,12 +11,11 @@ use shuttle_axum::ShuttleAxum;
 use tracing::info;
 
 mod logger;
-use logger::Logger;
 
 mod state;
 use state::AppState;
 
-#[shuttle_runtime::main(tracing_layer = Logger::make_layer)]
+#[shuttle_runtime::main(tracing_layer = logger::Logger::make_layer)]
```
Comment (iulianbarbu) [inline tracing/axum-logs-endpoint/src/main.rs:18]: Is this going to work after we merge the logger? I think the custom layer will not be supported anymore.

Comment (oddgrd) [reply tracing/axum-logs-endpoint/src/main.rs:18]: Indeed, the custom tracing layer is no longer supported. Users who want to customize their tracing setup can just opt out of the default now and set it up however they want. I think we can just remove them for now.

Comment (orhun) [reply tracing/axum-logs-endpoint/src/main.rs:18]: Should we remove this example then? Or do you mean the `custom-layer` example?

Comment (orhun) [reply tracing/axum-logs-endpoint/src/main.rs]: Removed the examples in 679f476ec9dd9083269c18c6d9f6885db9962411

**Pull request merged**